File: solution_set1/2269. Find the K-Beauty of a Number.py
##ss
class Solution:
    def divisorSubstrings(self, num: int, k: int) -> int:
        
        count = 0
        
        num = str(num)
        
        for x in range(len(num)-k+1):
            temp = num[x:x+k]
            
            if int(temp)!=0 and int(num) % int(temp) == 0:
                count+=1
                
        return count
        

File: cpp_solutions/2269. Find the K-Beauty of a Number.cpp
class Solution {
public:
    int divisorSubstrings(int num, int k) {
        int count = 0;
        
        
        auto revs = std::to_string(num);
        for(int x=0;x<revs.size()-k+1;x++){
            ///cout << revs.substr(x,x+k)<<"\n";
            auto temp = revs.substr(x,k);
            int comp = stoi(temp);
            //cout << comp << '\n' << x << "--" << x+k << '\n';
            if (comp!=0 && num%comp==0){
                count++;
            } 
        }
        
        return count;
        
    }
};
